### PR TITLE
Add filter readyinminutes - WITH ANTHONY AND CJ 

### PR DESCRIPTION
**Pull request by techprojsforfun**
*PAIR PROGRAM WITH ANTHONY AND CJ* 
add ability to filter based on preparation time

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -91,11 +91,86 @@ def set_vars_from_curr_user
 		@cuisine = current_user.cuisine
 	end
 
-
+	@all_recipes = {'items' =>
```
Comment (ChloeGuyue) [inline app/controllers/users_controller.rb:94]: We should remember to remove @all_recipes if necessary as refactoring in the next iteration.

File: app/models/recipe.rb [hunk 2]
```diff
@@ -1,63 +1,40 @@
...
-        res_json_hash = Recipe.restructure_hash(res_json_hash, all_ids)
-        get_recipe_info_params = ''
-        all_ids.each_with_index do |elem, idx|
-            get_recipe_info_params = Recipe.concatenate_params(elem, idx, get_recipe_info_params)
-            # get recipe nutrition
-            get_recipe_nutrition_uri = URI(base_uri + '/recipes/' + elem.to_s + '/nutritionWidget.json')
-            res_recipe_nutrition_json_hash = call_api(get_recipe_nutrition_uri)
-            res_json_hash['items'][idx]['calories'] = res_recipe_nutrition_json_hash['calories'].to_i
+    @have_slept = false
+    def self.find_in_api(calories, budget, time)
```
Comment (ChloeGuyue) [inline app/models/recipe.rb:13]: This method looks good.

File: app/models/recipe.rb [hunk 3]
```diff
@@ -1,63 +1,40 @@
 require 'net/http'
 class Recipe < ActiveRecord::Base
     has_and_belongs_to_many :users
-    def self.restructure_hash(res_json_hash, all_ids)
```
Comment (nkompella) [inline app/models/recipe.rb:4]: The refactoring of this method looks really good! Makes everything far cleaner and more concise. Well done!

File: features/add_favorite.feature [hunk 4]
```diff
@@ -17,6 +17,7 @@ Feature: Add favorite recipes
         And I fill in "age" with "18"
         And I fill in "weight" with "50"
         And I select "60" from "height"
+        And I fill in "Time available to cook per day" with "60"
```
Comment (nkompella) [inline features/add_favorite.feature:20]: I'm glad you update tests for everything!

File: app/models/recipe.rb (hunk 2 above)
Comment (nkompella) [reply app/models/recipe.rb:13]: I agree!

**Pull request merged**